Python/chapter03/p01_three_in_one/miguelHx.py

```diff
@@ -0,0 +1,420 @@
+"""
+Python version 3.7.0
+3.1 - Three in one
+Describe how you could use a single array to implement three stacks
+"""
+
+import copy
+import unittest
+from typing import List
+
+
+class StackNode(object):
+
+    def __init__(self, data):
+        self.data = data
+        self.next = None # next is a pointer to a StackNode object
+
+class MyStack(object):
```

Comment:
seems like this class is not being used in the solution of the problem (StackTrio)

Reply:
It is not.  But StackTrio uses StackNode, which MyStack also uses.  I figured that if changes would potentially need to be made to StackNode, then I'll make updates to both MyStack and StackTrio in the same file.

The reason why I did not leave MyStack in a separate file and import it is because I ran into problems with relative imports.

Reply:
oh yes, relative imports don't work _at all_ outside of packages :(

most codebases tend to prefer removing code that is not currently used, and deferring its introduction to the moment where they start being used. in this case, the `MyStack` class could safely be removed from this solution and appear only in 3.2.

`StackNode` can be kept.

Reply:
Updated here: 098292e6fb4390059ef567a031e45053bee74735